ui/src/router/components/form/components/experiment_config/EngineTypePanel.js

```diff
@@ -4,8 +4,10 @@ import { EuiForm, EuiFormRow, EuiSuperSelect } from "@elastic/eui";
 import { DescribedFormGroup } from "../../../../../components/form/described_form_group/DescribedFormGroup";
 import { FormLabelWithToolTip } from "../../../../../components/form/label_with_tooltip/FormLabelWithToolTip";
 
-export const EngineTypePanel = ({ type, options, onChange, errors }) => {
+export const EngineTypePanel = ({ type, options, protocol, onChange, errors }) => {
   const selectedOption = options.find((option) => option.value === type);
+  const helpText = protocol === "UPI_V1" ? 
```

Comment:
I feel like this help text here can be moved to the field source dropdown tooltip - when UPI is chosen as the protocol, "payload" should not even be visible as an option both for traffic rules and the experiment engine variables config, right? For the custom experiment engines, this actually comes from the experiment engine's microfrontend ([ref](https://github.com/caraml-dev/turing/blob/main/engines/experiment/docs/developer_guide.md#ui-1)) which I believe is currently missing now and should be added to the XP repo, for Turing Experiments. Is this TBD?

Reply:
> "payload" should not even be visible as an option both for traffic rules and the experiment engine variables config, right? 

Yes, it is as such for traffic rules now

> For the custom experiment engines, this actually comes from the experiment engine's microfrontend ([ref](https://github.com/caraml-dev/turing/blob/main/engines/experiment/docs/developer_guide.md#ui-1)) which I believe is currently missing now and should be added to the XP repo, for Turing Experiments. Is this TBD?

No, it was intentional as I thought the experiment engine doesn't need to know about the existence of UPI as the interaction with experiment engine required no changes. If shifted, all experiment engine would have to modify the remote ui to display on HTTP. Also, will this will lead to requiring experiment engine to be explicit about supporting UPI ? They should be able to opt to not support this if they don't use UPI.

This said I do agree there is some inconsistency with the traffic rules. There so much work involved just to add a tooltip below 😰  I will address this in a separate PR if that is fine.

Reply:
removed and opened [PR](https://github.com/caraml-dev/xp/pull/49) in XP